packages/core/user-profile/core-user-profile-browser-internal/src/user_profile_service.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Logger } from '@kbn/logging';
+import type { CoreContext, CoreService } from '@kbn/core-base-browser-internal';
+import type { CoreUserProfileDelegateContract } from '@kbn/core-user-profile-browser';
+import type {
+  InternalUserProfileServiceSetup,
+  InternalUserProfileServiceStart,
+} from './internal_contracts';
+import { getDefaultUserProfileImplementation, convertUserProfileAPI } from './utils';
+
+export class UserProfileService
+  implements CoreService<InternalUserProfileServiceSetup, InternalUserProfileServiceStart>
+{
+  private readonly log: Logger;
+  private delegate?: CoreUserProfileDelegateContract;
```

Comment:
`userProfile` service implementation follow the same idea as the security one: register the delegate, convert it, re-expose it.